src/lib/common/sol-certificate-impl-linux.c

```diff
@@ -147,3 +148,25 @@ sol_cert_get_filename(const struct sol_cert *cert)
 
     return cert->filename;
 }
+
+SOL_API struct sol_blob *
+sol_cert_get_contents(const struct sol_cert *cert)
+{
+    struct sol_blob *blob;
+    size_t size;
+    char *data;
+
+    SOL_NULL_CHECK(cert, NULL);
+
+    data = sol_util_load_file_string(cert->filename, &size);
+    SOL_NULL_CHECK(data, NULL);
+
+    blob = sol_blob_new(SOL_BLOB_TYPE_DEFAULT, NULL, data, size);
+    if (!blob) {
+        SOL_WRN("Could not allocate memory for sol_blob");
+        free(data);
+        return NULL;
+    }
```

Comment:
btw, just remembered that in `src/modules/flow/file/file.c`  we have `file_reader_blob_type`. We should make that a public one in sol-util-file.h and share it here. It would be more efficient than using `sol_util_load_file_string()` since if possible it wouldn't copy the contents to memory, rather use mmap()